### PR TITLE
Add Ajax handler and postbox for auto-command

### PR DESCRIPTION
**Pull request by JoosuaKoskinen**
#### What are the main changes in this PR?

Adds AJAX handler and pre-made postbox for automatically sending AJAX request and running a command on page load.

Also fixes a couple of bugs.

Example usage here:
https://github.com/Seravo/seravo-plugin/blob/postbox/upgrade-backups-page/modules/backups.php#L35-L53

### CODE REVIEW

File: lib/postbox/ajax.php [hunk 1]
```diff
@@ -287,5 +287,131 @@ public static function exception_response() {
...
+     * @param bool $allow_failure Whether exit code other than 0 should respond with an error.
+     */
+    public function __construct( $section, $command = null, $cache_time = 300, $allow_failure = false ) {
+      parent::__construct($section);
+
+      $this->command = $command;
+      $this->allow_failure = $allow_failure;
+
+      $this->set_build_func(
+        function ( \Seravo\Postbox\Component $base, $section ) {
```
Comment (Moppa5) [inline lib/postbox/ajax.php:342]: Here again, are this some rector related case?

File: rector.php [hunk 2]
```diff
@@ -211,7 +211,7 @@
 
   // Type Declaration
   $services->set(\Rector\TypeDeclaration\Rector\ClassMethod\AddArrayReturnDocTypeRector::class);
-  $services->set(\Rector\TypeDeclaration\Rector\Property\CompleteVarDocTypePropertyRector::class);
+  //$services->set(\Rector\TypeDeclaration\Rector\Property\CompleteVarDocTypePropertyRector::class); Rector being Rector
```
Comment (Moppa5) [inline rector.php:214]: Mystical comments :D

**Review by Moppa5 — CHANGES_REQUESTED** [review]
Great job! I didn't test the new stuff quite extensively but test widgets on backup page seem to work normally and there are no major flaws. Possible arising issues can be fixed when converting the widgets as well.

However, I really think we should reconsider where we put the `extends PostBox` and similar. If we have for example 5 ajax commands, the file becomes quickly quite large and it gets difficult to grasp the contents of the file. I don't think there's any issue for example having a individual file for every module that extends Postbox or Ajax_Handler. What do you think about this? Personally I think it's not good in the terms of maintainability.

File: lib/postbox/component.php [hunk 3]
```diff
@@ -53,9 +53,9 @@ public function add_children( $children ) {
       }
 
       foreach ( $children as $child ) {
-        if ( $child instanceof Component ) {
-          $this->children[] = $child;
-        }
+        // Add them with add_child so same checks
+        // are used for each child.
+        $this->add_child($child);
```
[the remark this answers is not in the document]
Comment (JoosuaKoskinen) [reply lib/postbox/component.php:58]: It is and I still call it false positive. But it was quite useless check so removed it to make Rector happy.

File: rector.php (hunk 2 above)
Comment (JoosuaKoskinen) [reply rector.php:214]: Fixed

File: lib/postbox/ajax.php (hunk 1 above)
Comment (JoosuaKoskinen) [reply lib/postbox/ajax.php]: Fixed

File: lib/postbox/ajax.php [hunk 4]
```diff
@@ -287,5 +287,131 @@ public static function exception_response() {
...
+      $response->set_error($error);
+      return $response;
+    }
+
+  }
+}
+
+namespace Seravo\Postbox\Ajax;
+
+use \Seravo\Postbox\Ajax_Handler;
```
[the remark this answers is not in the document]
Comment (JoosuaKoskinen) [reply lib/postbox/ajax.php:310]: Fixed.

**Review by Moppa5 — APPROVED** [review]
